src/test/java/com/arturfrimu/exercisesback/service/ExerciseGenerationServiceImplTest.java

```diff
@@ -23,20 +23,22 @@
 class ExerciseGenerationServiceImplTest {
 
     @Autowired
-    private ExerciseGenerationServiceImpl exerciseGenerationService;
+    private ExerciseManagementService exerciseManagementService;
+    @Autowired
+    private FindPercentageExerciseService findPercentageExerciseService;
 
     @BeforeEach
     public void setup() {
         MockitoAnnotations.openMocks(this);
-        exerciseGenerationService.clear();
+        exerciseManagementService.clear();
```

Comment:
Nu prea vad sens in clasa ExerciseManagementService

Injecteaza drept DAO unde ai nevoie de metodele clear si putAll si s-a rezolvat problema